### PR TITLE
Amrfinderplus tool

### PR DESCRIPTION
**Pull request by pimarin**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection

Amrfinderplus wrapper linked to the already existing amrfinder data_manager

### CODE REVIEW

File: tools/amrfinderplus/amrfinderplus.xml [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+    <command detect_errors="aggressive"><![CDATA[
+        amrfinder
+        #*======================================
+                    CPU option
+        ======================================*#
+        --threads \${GALAXY_SLOTS:-1}
+        #*======================================
+                    DATABASE
+        ======================================*#
+        --database "$input_option.amrfinder_db_select.fields.path"
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/amrfinderplus/amrfinderplus.xml:21]: and everywhere else

File: tools/amrfinderplus/amrfinderplus.xml [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+                </conditional>
+            </section>
+            <output name="amrfinderplus_report" value="test_8/amrfinderplus_results.tsv" ftype="tabular" lines_diff="1"/>
+            <output name="protein_output" value="test_8/amrfinderplus_protein_output.fasta" ftype="fasta" lines_diff="1"/>
+            <output name="nucleotide_output" value="test_8/amrfinderplus_nucleotide_output.fasta" ftype="fasta" lines_diff="1"/>
+            <output name="mutation_all_report" value="test_8/amrfinderplus_mutation_all.tsv" ftype="tabular" lines_diff="1"/>
+        </test>   
+    </tests>
+    <help><![CDATA[**What it does**
+        devfvsrv
```
Comment (bgruening) [inline tools/amrfinderplus/amrfinderplus.xml:368]: :) this needs some :heart:

Comment (pimarin) [reply tools/amrfinderplus/amrfinderplus.xml:368]: XD my bad, to much happy to submit to the IUC !

File: tools/amrfinderplus/amrfinderplus.xml [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+                    <param name="annotation_format" value="prokka"/>
+                </conditional>
+            </section>
+            <output name="amrfinderplus_report" value="test_6/amrfinderplus_results.tsv" ftype="tabular" lines_diff="1"/>
+            <output name="protein_output" value="test_6/amrfinderplus_protein_output.fasta" ftype="fasta" lines_diff="1"/>
+        </test>   
+        <test expect_num_outputs="2"> <!-- TEST_7 protein input and genbank default annotation  -->
+            <section name="input_option">
+                <param name="amrfinder_db_select" value="V3.11.1-2023-04-17.1"/>
+                <conditional name="input_mode">
```
Comment (bgruening) [inline tools/amrfinderplus/amrfinderplus.xml:332]: why are those conditionals? and not pure select boxes?

File: tools/amrfinderplus/amrfinderplus.xml [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+                    <param name="gff_annotation" value="annotation_ncbi.gff3"/>
+                </conditional>
+            </section>
+            <output name="amrfinderplus_report" value="test_7/amrfinderplus_results.tsv" ftype="tabular" lines_diff="1"/>
+            <output name="protein_output" value="test_7/amrfinderplus_protein_output.fasta" ftype="fasta" lines_diff="1"/>
+        </test>   
+        <test expect_num_outputs="4"> <!-- TEST_8 both protein and nucleotide input  -->
+            <section name="input_option">
+                <param name="amrfinder_db_select" value="V3.11.1-2023-04-17.1"/>
+                <conditional name="input_mode">
```
Comment (bgruening) [inline tools/amrfinderplus/amrfinderplus.xml:344]: select box?

File: tools/amrfinderplus/amrfinderplus.xml [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+                    <param name="input_select" value="nucl_prot"/>
+                    <param name="protein_input" value="enterococcus_faecalis.faa"/>
+                    <param name="nucleotide_input" value="enterococcus_faecalis.fna"/>
+                    <param name="gff_annotation" value="enterococcus_faecalis_bakta.gff3"/>
+                    <param name="annotation_format" value="bakta"/>
+                </conditional>
+            </section>
+            <section name="organism_options">
+                <conditional name="organism_conditionnal">
```
Comment (bgruening) [inline tools/amrfinderplus/amrfinderplus.xml:353]: select?

File: tools/amrfinderplus/tool-data/amrfinderplus_database.loc.sample [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# this is a tab separated file describing the location of amrfinderplus database
```
Comment (bgruening) [inline tools/amrfinderplus/tool-data/amrfinderplus_database.loc.sample:1]: I think this file needs to be named `.test` isn't it?

Comment (bgruening) [reply tools/amrfinderplus/tool-data/amrfinderplus_database.loc.sample:1]: Nope I'm wrong https://github.com/galaxyproject/tools-iuc/blob/main/tools/bowtie2/tool-data/bowtie2_indices.loc.sample

File: tools/amrfinderplus/amrfinderplus.xml (hunk 3 above)
Comment (pimarin) [reply tools/amrfinderplus/amrfinderplus.xml:332]: you right, I'll change it !

File: tools/amrfinderplus/amrfinderplus.xml (hunk 4 above)
Comment (pimarin) [reply tools/amrfinderplus/amrfinderplus.xml:344]: To the database version ? Only one DB could be used. Select box is available with `multiple="false"`. there are numbersof amrfinder database version

File: tools/amrfinderplus/amrfinderplus.xml (hunk 5 above)
Comment (pimarin) [reply tools/amrfinderplus/amrfinderplus.xml:353]: Some option are only available when an organism is provided, with the conditional, I can suggest this option

File: tools/amrfinderplus/amrfinderplus.xml (hunk 3 above)
Comment (pimarin) [reply tools/amrfinderplus/amrfinderplus.xml:332]: hum as for organism some options are only available if I have nucleotide or protein provided

Comment (pimarin) [reply tools/amrfinderplus/amrfinderplus.xml:332]: I can make a radio button instead

File: tools/amrfinderplus/amrfinderplus.xml (hunk 4 above)
Comment (bgruening) [reply tools/amrfinderplus/amrfinderplus.xml:344]: Sorry, forget all my comments. I will start again.

File: tools/amrfinderplus/tool-data/amrfinderplus_database.loc.sample (hunk 6 above)
Comment (pimarin) [reply tools/amrfinderplus/tool-data/amrfinderplus_database.loc.sample:1]: I don't watch other tool with `.test` to the end of the tool-data file

**Review by bgruening — APPROVED** [review]
Looks great to me. I was just confused - sorry for the noise. Thanks for the great help section.

**Pull request merged**